src/SocialShare/Provider/Tumblr.php

```
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the SocialShare package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SocialShare\Provider;
+
+class Tumblr implements ProviderInterface
+{
+    const NAME = 'tumblr';
+    const SHARE_URL = 'http://www.tumblr.com/share/link?%s';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getLink($url, array $options = array())
+    {
+        $options['url'] = $url;
+
+        return sprintf(self::SHARE_URL, http_build_query($options, null, '&'));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getShares($url)
+    {
+        // This information cannot be retrieved from Tumblr API
+        return 0;
```

Maybe than trowing an exception is better than silently returning 0?